### PR TITLE
Add a catch all server declaration logic based on NGINX_DEFAULT_SERVER_NAME

### PR DESCRIPTION
**Pull request by andypanix**

### CODE REVIEW

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -3,7 +3,15 @@ set -e
...
-export NGINX_DEFAULT_SERVER_NAME=${NGINX_DEFAULT_SERVER_NAME:-drupal}
+# As default we have an "_" as server name.
+export NGINX_DEFAULT_SERVER_NAME=${NGINX_DEFAULT_SERVER_NAME:-_}
+# If you want to declare the default.conf server as the default
+# catch-all server, set the env var DECLARE_DEFAULT_SERVER to 1.
+if [ $DECLARE_DEFAULT_SERVER == 1 ]; then
+  export DEFAULT_SERVER="default_server"
+else
+  export DEFAULT_SERVER=""
+fi
```
Comment (paolomainardi) [inline docker-entrypoint.sh]: we could infere the `DEFAULT_SERVER` by checking if `NGINX_DEFAULT_SERVER_NAME == '_'`

Comment (paolomainardi) [reply docker-entrypoint.sh]:
```
if [ $NGINX_DEFAULT_SERVER_NAME == '_']; then
  export DEFAULT_SERVER="default_server"
else
  export DEFAULT_SERVER=""
fi
```

**Comment by paolomainardi** [comment]
LGTM :+1:

**Comment by alessiopiazza** [comment]
👍

File: templates/catch-all-server.conf [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# not served by others server block declarations.
+server {
+    listen       80 default_server;
+    # The _ it is just one of a myriad of invalid domain
+    # names which never intersect with any real name.
+    server_name  _;
+    # 444 error code: no response, instruct the server to return no
+    # information to the client and close the connection immediately.
+    return       444;
+}
```
Comment (paolomainardi) [inline templates/catch-all-server.conf]: I have a doubt here, if the files are loaded in an alphabetical order and I add a file called 000-redirect.conf from a custom Docker image, it should behave then like the catch-all server.

**Pull request merged**